packages/java/endpoint/src/main/java/dev/hilla/crud/filter/Filter.java

```diff
@@ -0,0 +1,13 @@
+package dev.hilla.crud.filter;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "t")
+@JsonSubTypes({ @Type(value = OrFilter.class, name = "o"),
+        @Type(value = AndFilter.class, name = "a"),
+        @Type(value = PropertyStringFilter.class, name = "p") })
+public interface Filter {
```

Comment:
I would consider adding a `type` property to `Filter`, and then using an `Enum` for the values, if only so that we get the enum generated on the client and developers don't have to use strings and can actually use the `Filter` type and sub-types when defining filters. I've seen https://github.com/vaadin/hilla/issues/438, which would be a better solution, but this seems like an easy improvement in the meantime.

Reply:
As discussed let's postpone this for now, to either improve the generator to support `@JsonSubTypes` or create a custom TS API.